manual/trees/index.md

```diff
@@ -7,985 +7,543 @@ toc: true
+"Friend trees" are added by calling [TTree::AddFriend()](https://root.cern/doc/master/classTTree.html#a321f2684de145cfcb01cabfce69ea910){:target="_blank"}.
+Adding another tree called `T1` as a friend tree will make the branch `X` of `T1` available as both `T1.X` and - if `X` does not exist in the original tree - as `X`.
 
-{% include figure_jsroot
-   file="trees.root" object="c3" width="500px" height="350px"
-   caption="The variable `Cost` and `Age` with a selection drawn in a histogram."
-%}
+Friend trees are expected to have at least as many entries as the original tree.
+The order of the friend tree's entries must preserve the entry order of the original tree.
 
-Next, the fourth pad is activated and the histogram is drawn with the draw option `surf2`.
-Refer to the {% include ref class="THistPainter" %} class for possible draw options.
+> **Note**
+>
+> Care must be taken to ensure that the order of entries in the primary tree matches friends' entries. This is especially relevant when processing a tree in parallel to generate a friend tree, as the entries might be written out in an undefined order (misaligned entries).
 
 {% highlight C++ %}
-   root [] myCanvas->cd(4)
-   root [] MyTree->Draw("Cost:Age","Nation == \"CH\"","colz")
-{% endhighlight %}
+void treeWithFriend() {
+   std::unique_ptr<TFile> myFile( TFile::Open("file.root") );
+   auto tree = myFile->Get<TTree>("TreeName");
 
-{% include figure_jsroot
-   file="trees.root" object="c4" width="500px" height="350px"
-   caption="The variable `Cost` and `Age` with a selection and a draw option drawn in a histogram."
-%}
-
-The [TTree::Draw()](https://root.cern/doc/master/classTTree.html#ac4016b174665a086fe16695aad3356e2){:target="_blank"} method also accepts {% include ref class="TCut" %} objects. A {% include ref class="TCut" %} object is
-a specialized string object used for {% include ref class="TTree" %} selections.
+   std::unique_ptr<TFile> myFriendFile( TFile::Open("friend.root") );
+   auto friendTree = myFriendFile->Get<TTree>("FriendTreeName");
 
+   tree->AddFriend(friendTree);
 
-### Using TTree::MakeClass()
+   int variable;
+   tree->SetBranchAddress("branchName", &variable);
+   int variableFriend;
+   tree->SetBranchAddress("FriendTreeName.friendBranchName", &variableFriend);
 
-- Use the [TTree::MakeClass()](https://root.cern/doc/master/classTTree.html#ac4ceaf4ae0b87412acf94093043cc2de){:target="_blank"}
-method, to generate a skeleton class for looping over the entries of a tree.
+   // Iteration over `tree` automatically advances its friend trees.
+   for (int iEntry = 0; tree->LoadTree(iEntry) >= 0; ++iEntry) {
+      // Load the data for the given tree entry
+      tree->GetEntry(iEntry);
 
+      printf("%d %d\n", variable, variableFriend);
+   }
+{% endhighlight %}
 
-### Using TTree::MakeSelector()
+# We still need to work on the parts below. No point in reviewing yet.
```

Comment:
If this is going to be merged in its current state, please remove this!
```suggestion
```

Reply:
Nice catch thanks!